appservice/src/deploy/runWithZipStream.ts

```diff
@@ -34,8 +34,17 @@ export async function runWithZipStream(context: IActionContext, fsPath: string,
         });
     } else {
         ext.outputChannel.appendLog(localize('zipCreate', 'Creating zip package...'), { resourceName: client.fullName });
-        const zipFile = new yazl.ZipFile();
+        const zipFile: yazl.ZipFile = new yazl.ZipFile();
         let filesToZip: string[] = [];
+        let sizeOfZipFile: number = 0;
+
+        zipFile.outputStream.on('data', (chunk) => {
+            if (Array.isArray(chunk)) {
```

Comment:
It looks like it's possible for chunk to be a string or buffer as well: https://nodejs.org/api/stream.html#stream_event_data

Reply:
Hmm, my understanding was Buffers are a TypedArray which are in term a type of array.  Might be safer to just do a check on length, but the issue is chunk is `any` so this how I was type-checking it.

Reply:
I don't like using `unknown[]` because I don't think that's technically accurate and I do like `if (Array.isArray(chunk)) {`. What was wrong with that? Can we change back?

Reply:
I can change it back  I just did this because in the yazl types, the return type of chunk is actually `any[]` so I don't think `unknown[]` is technically inaccurate.
```
class EventEmitter {
        addListener(event: string | symbol, listener: (...args: any[]) => void): this;
        on(event: string | symbol, listener: (...args: any[]) => void): this;
```

Reply:
`(...args: any[])` is different than `(args: any[])`

The former says all the args combined are an array, the latter says just the first arg is an array

Reply:
Oh, doh you are totally right.